### PR TITLE
docs Fourth cherry pick 2.5

### PR DESCRIPTION
**Pull request by angelinalg**
https://github.com/ray-project/ray/pull/35816
https://github.com/ray-project/ray/pull/36054
https://github.com/ray-project/ray/pull/36112

## Why are these changes needed?

observability docs refactor
cc: @zhe-thoughts @ArturNiederfahrenhorst @scottsun94 @rkooo567 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by zhe-thoughts — APPROVED** [review]
Clean docs changes. OK to pick once CI all pass

**Comment by ArturNiederfahrenhorst** [comment]
Doc tests are generally not stable at the moment but the errors I see on this PR seem related to these changes here.
Please check!

File: python/ray/runtime_context.py [hunk 1]
```diff
@@ -360,16 +360,25 @@ def _get_actor_call_stats(self):
...
         .. testcode::
 
             import ray
             # Get the job id.
             ray.get_runtime_context().get_job_id()
+            # Get the actor id.
+            ray.get_runtime_context().get_actor_id()
+            # Get the task id.
+            ray.get_runtime_context().get_task_id()
+            # Get the worker id.
```
Comment (angelinalg) [inline python/ray/runtime_context.py:380]:
```suggestion

```

File: python/ray/runtime_context.py [hunk 2]
```diff
@@ -360,16 +360,25 @@ def _get_actor_call_stats(self):
...
+            ray.get_runtime_context().get_worker_id()
```
Comment (angelinalg) [inline python/ray/runtime_context.py:381]:
```suggestion

```

File: python/ray/runtime_context.py (hunk 1 above)
Comment (angelinalg) [inline python/ray/runtime_context.py:380]:
```suggestion

```

**Pull request closed without merging**